Plugins/EDM4hep/include/ActsPlugins/EDM4hep/EDM4hepUtil.hpp

```diff
@@ -32,6 +32,22 @@
 #include <edm4hep/SimTrackerHit.h>
 #include <edm4hep/Track.h>
 #include <edm4hep/TrackState.h>
+#include <podio/podioVersion.h>
+
+#if podio_VERSION_MAJOR == 0 || \
+    (podio_VERSION_MAJOR == 1 && podio_VERSION_MINOR <= 2)
+
+template <>
+struct std::hash<podio::ObjectID> {
+  std::size_t operator()(const podio::ObjectID& id) const noexcept {
+    auto hash_collectionID = std::hash<std::uint32_t>{}(id.collectionID);
+    auto hash_index = std::hash<int>{}(id.index);
+
+    return hash_collectionID ^ hash_index;
+  }
+};
+
+#endif
```

Comment:
this polyfills for older versions of podio right?

I just wondered if this could be pushed to the source file somehow. but putting just the method declaration would still count as a double definition for the `struct` I guess?
```c++
template <>
struct std::hash<podio::ObjectID> {
  std::size_t operator()(const podio::ObjectID& id) const noexcept;
};
```

anyways, just a thought

Reply:
Yeah, they added a specialization only in 1.3. I didn't put it in the cpp file but I had it in the bottom of the header before, and that tripped up at least on one compiler (I don't remember which one). I can try again if you want.